src/Elastic.Markdown/Slices/HtmlWriter.cs

```diff
@@ -12,29 +14,67 @@ public class HtmlWriter(DocumentationSet documentationSet, IFileSystem writeFile
 {
 	private DocumentationSet DocumentationSet { get; } = documentationSet;
 
-	private async Task<string> RenderNavigation(MarkdownFile markdown, Cancel ctx = default)
+	private async Task<string> RenderNavigation(ConfigurationFile configuration, string topLevelGroupId, MarkdownFile markdown, Cancel ctx = default)
 	{
+		var group = DocumentationSet.Tree.NavigationItems
+			.OfType<GroupNavigation>()
+			.FirstOrDefault(i => i.Group.Id == topLevelGroupId)?.Group;
+
 		var slice = Layout._TocTree.Create(new NavigationViewModel
 		{
-			Tree = DocumentationSet.Tree,
-			CurrentDocument = markdown
+			Title = group?.Index?.NavigationTitle ?? DocumentationSet.Tree.Index!.NavigationTitle!,
+			TitleUrl = group?.Index?.Url ?? DocumentationSet.Tree.Index!.Url,
+			Tree = group ?? DocumentationSet.Tree,
+			CurrentDocument = markdown,
+			IsRoot = topLevelGroupId == DocumentationSet.Tree.Id,
+			Configuration = configuration
```

Comment:
Since we only use this in `GetHxSelectOob` to get the features, set the features on the model and not the whole `Configuration`?

Reply:
https://github.com/elastic/docs-builder/pull/604/commits/ee10395c27ec65fd6c11127bacec42c344aa0a80